### PR TITLE
Improved nested code

### PR DESCRIPTION
**Pull request by eusonlito**
I think tha I have improved some nested ifs. There are some places that I have changed `!isset` with `empty` that could be de same in this context and best than negation boolean.

Also splited `val` function into `valString` and `valArray` and added an Exception is not array or string.

Accept pull if you think that are this patches are improvements :)

### CODE REVIEW

File: src/Containers/CollectionMultiple.php [hunk 1]
```diff
@@ -57,7 +57,7 @@ public function getTemplate($index = '::n::')
      */
     public function pushVal($value = null)
     {
-        if (!isset($value[$this->keyField])) {
+        if (empty($value[$this->keyField])) {
```
Comment (oscarotero) [inline src/Containers/CollectionMultiple.php]: The point here is to know whether or not this value is defined, not if it's empty or not. The values "0" or even "" are perfectly valid.

File: src/Containers/CollectionMultiple.php [hunk 2]
```diff
@@ -76,7 +76,7 @@ public function pushVal($value = null)
      */
     public function pushLoad($value = null, $file = null)
     {
-        if (!isset($value[$this->keyField])) {
+        if (empty($value[$this->keyField])) {
```
Comment (oscarotero) [inline src/Containers/CollectionMultiple.php]: The same comment in `pushVal` is valid here

File: src/Containers/Collection.php [hunk 3]
```diff
@@ -47,10 +47,12 @@ public function load($value = null, $file = null)
...
-            foreach ($value as $k => $v) {
-                $this->pushLoad($v, isset($file[$k]) ? $file[$k] : null);
-            }
+        if (empty($value)) {
+            return $this;
+        }
+
+        foreach ($value as $k => $v) {
+            $this->pushLoad($v, isset($file[$k]) ? $file[$k] : null);
         }
```
Comment (oscarotero) [inline src/Containers/Collection.php]: I don't see the benefits of this. You are returning `$this` in two places.

Comment (eusonlito) [inline src/Containers/Collection.php]: I know, but you're avoiding nesting code. More clear.

Comment (oscarotero) [inline src/Containers/Collection.php]: I don't think a function with two returns is more clear than with one. I can see this in big functions but here there is only one "if".

**Pull request merged**